### PR TITLE
Show LongDataDisplay controls only when needed

### PR DESCRIPTION
**Pull request by buberdds**
Slack request

https://d05adace.oasis-explorer.pages.dev/mainnet/sapphire/tx/0xb0846c1ad0d3279fbf70eec6cf8a51a9141a746d8b26a55a24292f2c7c3b4d5c
https://d05adace.oasis-explorer.pages.dev/mainnet/sapphire/tx/0x84a24df4c7f8af3363df798155ffe57135861e69d0677c0148fbd3a9010f1acb

vs

https://explorer.dev.oasis.io/mainnet/sapphire/tx/0xb0846c1ad0d3279fbf70eec6cf8a51a9141a746d8b26a55a24292f2c7c3b4d5c

### CODE REVIEW

File: src/app/components/LongDataDisplay/index.tsx [hunk 1]
```diff
@@ -1,47 +1,71 @@
...
       <Typography
-        variant="mono"
-        sx={{
+        ref={textRef}
+        style={{
           fontWeight,
+          maxHeight: isExpanded ? 'none' : collapsedContainerMaxHeight,
+          overflow: 'hidden',
+          display: 'flex',
+          lineClamp: isExpanded ? 'none' : collapsedLinesNumber,
```
Comment (lukaw3d) [inline src/app/components/LongDataDisplay/index.tsx:58]: This line is completely ignored. If you can use
```css
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;  
  overflow: hidden;
```
it should work and show `...` at end

File: src/app/components/LongDataDisplay/index.tsx [hunk 2]
```diff
@@ -1,47 +1,70 @@
...
-          sx={{
-            fontWeight,
-            overflowWrap: 'anywhere',
-          }}
-        >
-          {data}
-        </Typography>
-      </Collapse>
-      {data.length > threshold && (
-        <Link sx={{ cursor: 'pointer' }} onClick={() => setShowData(!showData)}>
```
[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/app/components/LongDataDisplay/index.tsx:42]: Minor styling issue: due to our default theme overrides the focused state looks funny https://github.com/oasisprotocol/explorer/blob/8b6e99f9dbc1ac79414fe88b7ba361c74d80968e/src/styles/theme/defaultTheme.ts#L219-L220

image

**Review by lukaw3d — APPROVED** [review]

File: src/app/components/LongDataDisplay/index.tsx (hunk 1 above)
Comment (lukaw3d) [reply src/app/components/LongDataDisplay/index.tsx]: Aw do we not want `...`?

Comment (buberdds) [reply src/app/components/LongDataDisplay/index.tsx]: with show button it's clear that data is trimmed. will ask Don

File: src/app/components/LongDataDisplay/index.tsx [hunk 3]
```diff
@@ -1,47 +1,81 @@
...
         variant="mono"
+        ref={textRef}
         sx={{
           fontWeight,
+          maxHeight: isExpanded ? 'none' : collapsedContainerMaxHeight,
+          overflow: 'hidden',
+          lineHeight: `${lineHeight}px`,
           overflowWrap: 'anywhere',
+          display: '-webkit-box',
+          '-webkit-line-clamp': isExpanded ? 'none' : `${collapsedLinesNumber}`, // keep it as string to avoid MUI adding px
```
Comment (buberdds) [inline src/app/components/LongDataDisplay/index.tsx:69]: TODO: fix warning in console

**Pull request merged**